util/oidc/oidc.go

```diff
@@ -135,17 +135,21 @@ func NewClientApp(settings *settings.ArgoCDSettings, dexServerAddr string, dexTl
 	return &a, nil
 }
 
-func (a *ClientApp) oauth2Config(scopes []string) (*oauth2.Config, error) {
+func (a *ClientApp) oauth2Config(r *http.Request, scopes []string) (*oauth2.Config, error) {
```

Comment:
If only the url is necessary to find the config, pass only the URL instead of the whole request object.

```suggestion
func (a *ClientApp) oauth2Config(request *url.URL, scopes []string) (*oauth2.Config, error) {
```